samcli/lib/sync/infra_sync_executor.py

```diff
@@ -0,0 +1,370 @@
+    """
+
+    _build_context: BuildContext
+    _package_context: PackageContext
+    _deploy_context: DeployContext
+    _code_sync_resources: Set[str]
+
+    def __init__(self, build_context: BuildContext, package_context: PackageContext, deploy_context: DeployContext):
+        """Constructs the sync for infra executor.
+
+        Parameters
+        ----------
+        build_context : BuildContext
+            BuildContext
+        package_context : PackageContext
+            PackageContext
+        deploy_context : DeployContext
+            DeployContext
+        """
+        self._build_context = build_context
+        self._package_context = package_context
+        self._deploy_context = deploy_context
+
+        self._code_sync_resources = set()
+
+        session = Session(profile_name=self._deploy_context.profile, region_name=self._deploy_context.region)
+        self._cfn_client = self._boto_client("cloudformation", session)
+        self._s3_client = self._boto_client("s3", session)
+
+    def _boto_client(self, client_name: str, session: Session):
+        """
+        Creates boto client
+
+        Parameters
+        ----------
+        client_name: str
+            The name of the client
+        session: boto3.Session
+            The session created using customer config
+
+        Returns
+        -------
+        Service client instance
+        """
+        return get_boto_client_provider_from_session_with_config(session)(client_name)
+
+    def _compare_templates(self, template_path: str, stack_name: str) -> bool:
+        """
+        Recursively conpares two templates, including the nested templates referenced inside
+
+        Parameters
+        ----------
+        template_path : str
+            The template location of the current template
+        stack_name : str
+            The CloudFormation stack name that the template is deployed to
+
+        Returns
+        -------
+        bool
+            Returns True if two templates are identical
+            Returns False if two templates are different
+        """
+        current_template = self.get_template(template_path)
+
+        if not current_template:
+            LOG.debug("Cannot obtain a working current template for template path %s", template_path)
+            return False
+
+        try:
+            last_deployed_template_str = self._cfn_client.get_template(
+                StackName=stack_name, TemplateStage="Original"
+            ).get("TemplateBody", "")
+        except ClientError as ex:
+            LOG.debug("Stack with name %s does not exist on CloudFormation", stack_name, exc_info=ex)
+            return False
+
+        last_deployed_template_dict = yaml_parse(last_deployed_template_str)
+
+        sanitized_current_template = copy.deepcopy(current_template)
+        sanitized_last_template = copy.deepcopy(last_deployed_template_dict)
+
+        sanitized_resources = self._remove_unnecessary_fields(sanitized_current_template)
```

Comment:
nit: should we update the method name to something related to `sanitation` given that we changed the variable names?